uberfire-extensions/uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/ext/preferences/shared/bean/BasePreference.java

```
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.preferences.shared.bean;
+
+import org.uberfire.mvp.Command;
+import org.uberfire.mvp.ParameterizedCommand;
+
+/**
+ * Interface that all preference beans should implement. It allows load and save operations
+ * when injecting it through CDI. The methods in this class are only really implemented by
+ * a generated preference bean, that will be the one injected by CDI.
+ * @param <T> The preference bean type implementing the interface.
+ */
+public interface BasePreference<T> extends Preference {
```

Prob. better to throw a runtime exception if the unimplemented version is called...i.e. load is called on the portable generated impl.